File: ClientRuntimes/Java/src/main/java/com/microsoft/rest/HttpOperationResponse.java
/**
 *
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 */

package com.microsoft.rest;

import org.apache.http.HttpRequest;
import org.apache.http.HttpResponse;

/**
 * Represents the base return type of all ServiceClient REST operations.
 *
 * @param <T> the type of Http response object
 */
public class HttpOperationResponse<T> {
    private HttpRequest request;
    private HttpResponse response;
    private T body;

    /**
     * Gets information about the associated HTTP request.
     *
     * @return the HTTP request
     */
    public HttpRequest getRequest() {
        return this.request;
    }

    /**
     * Sets the HTTP request.
     *
     * @param request the HTTP request
     */
    public void setRequest(HttpRequest request) {
        this.request = request;
    }

    /**
     * Gets information about the associated HTTP response.
     *
     * @return the HTTP response
     */
    public HttpResponse getResponse() {
        return this.response;
    }

    /**
     * Sets the HTTP response.
     *
     * @param request the HTTP response
     */
    public void setResponse(HttpResponse request) {
        this.response = response;
    }

    /**
     * Gets the HTTP response object.
     * 
     * @return the HTTP response object
     */
    public T getBody() {
        return this.body;
    }

    /**
     * Sets the HTTP response object.
     * 
     * @param body the HTTP response object
     */
    public void setBody(T body) {
        this.body = body;
    }
}

File: ClientRuntimes/CSharp/ClientRuntime/HttpOperationResponse.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Net.Http;

namespace Microsoft.Rest
{
    /// <summary>
    /// Represents the base return type of all ServiceClient REST operations.
    /// </summary>
    public class HttpOperationResponse<T> : HttpOperationResponse
    {
        /// <summary>
        /// Gets or sets the response object.
        /// </summary>
        public T Body { get; set; }
    }

    /// <summary>
    /// Represents the base return type of all ServiceClient REST operations without response body.
    /// </summary>
    public class HttpOperationResponse
    {
        /// <summary>
        /// Gets information about the associated HTTP request.
        /// </summary>
        public HttpRequestMessage Request { get; set; }

        /// <summary>
        /// Gets information about the associated HTTP response.
        /// </summary>
        public HttpResponseMessage Response { get; set; }
    }
}